titan/exposures/knowledge.py

```diff
@@ -0,0 +1,189 @@
+from typing import List, Dict
+
+import numpy as np  # type: ignore
+import networkx as nx  # type: ignore
+
+from . import base_exposure
+from .. import agent as ag
+from .. import population
+from .. import model
+from .. import utils
+
+
+class Knowledge(base_exposure.BaseExposure):
+
+    name: str = "knowledge"
+    stats: List[str] = ["knowledge_aware"]
+    """
+    Knowledge collects the following stats:
+
+    * knowledge_aware - number of agents with active knowledge
+    """
+
+    def __init__(self, agent: "ag.Agent"):
+        super().__init__(agent)
+
+        self.active = False
+        self.opinion = 0.0
+
+    def init_agent(self, pop: "population.Population", time: int):
+        """
+        Initialize the agent for this exposure during population initialization (`Population.create_agent`).  Called only on exposures that are enabled per the params.
+
+        Stochastically make agent aware, if aware, set the opinion from the params.
+
+        args:
+            pop: the population this agent is a part of
+            time: the current time step
+        """
+        knowledge_params = self.agent.location.params.knowledge
+        if pop.pop_random.random() < knowledge_params.init:
+            self.active = True
+
+            # TO_REVIEW - non aware agents always start at opinoin of 0?
```

Comment:
Is this still to review?

Reply:
yeah

Reply:
I don't think it makes a difference about whether they get an opinion at initialization or when they become aware model-wise but it makes more sense to initiate with 0 from a non-model perspective?